### PR TITLE
CORE-8365 Use default values when /secured/preferences is down

### PR DESCRIPTION
**Pull request by ash3rz**
This PR enables the UI to be more resilient when /secured/preferences is down.  Previously, the UI would not load if this callback failed.  Now, a default set of preferences will be used, the user will be informed, and the UI will continue loading.

### CODE REVIEW

File: de-lib/src/main/java/org/iplantc/de/client/models/UserSettings.java [hunk 1]
```diff
@@ -271,6 +272,20 @@ public Splittable asSplittable() {
         return ret;
     }
 
+    public void useDefaultValues(UserInfo instance) {
+        JSONObject defaults = new JSONObject();
+        JSONObject defaultHomeDir = new JSONObject();
+
+        JSONString id = new JSONString(instance.getHomePath());
+        JSONString path = new JSONString(instance.getHomePath());
```
Comment (psarando) [inline de-lib/src/main/java/org/iplantc/de/client/models/UserSettings.java]: Keep in mind that this `getHomePath` method can currently return `null`, if something goes wrong parsing the user info JSON in the `BootstrapCallback`.

I'm guessing that the `getHomePath` method will still be updated in another PR to return a default in this case? We will also be updating the bootstrap endpoint to return an error message instead of the user's home path when `data-info` is down.

Comment (psarando) [inline de-lib/src/main/java/org/iplantc/de/client/models/UserSettings.java]: For this PR, I think it's a good idea to use `getHomePath` here.

If the bootstrap endpoint doesn't return the user's real home path because `data-info` is down, then I think this `getHomePath` method can return `/iplant/home/<username>` for now (it can be updated in another PR).

That static part of the path (`/iplant/home/`) should be configurable if it isn't already.

**Comment by ash3rz** [comment]
Thanks for the review!  I'll merge this one now and create a separate ticket and PR for the scenario when data-info is down.
